src/test/java/org/apache/pirk/storm/KafkaStormIntegrationTest.java

```diff
@@ -0,0 +1,327 @@
+  }
+
+  private HashMap<String,Object> createKafkaProducerConfig()
+  {
+    String kafkaHostName = "localhost";
+    Integer kafkaPorts = 11111;
+    HashMap<String,Object> config = new HashMap<String,Object>();
+    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaHostName + ":" + kafkaPorts);
+    config.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+    config.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+
+    return config;
+  }
+
+  private void loadTestData(KafkaProducer producer)
+  {
+    for (JSONObject dataRecord : Inputs.createJSONDataElements())
+    {
+      logger.info("Sending record to Kafka " + dataRecord.toString());
+      producer.send(new ProducerRecord<String,String>(topic, dataRecord.toString()));
+    }
+  }
+
+  private void performEncryption() throws Exception
+  {
+    // ArrayList<String> selectors = BaseTests.selectorsDomain;
+    ArrayList<String> selectors = new ArrayList<>(Arrays.asList("s.t.u.net", "d.e.com", "r.r.r.r", "a.b.c.com", "something.else", "x.y.net"));
+    String queryType = Inputs.DNS_HOSTNAME_QUERY;
+
+    Paillier paillier = new Paillier(BaseTests.paillierBitSize, BaseTests.certainty);
+
+    nSquared = paillier.getNSquared();
+
+    queryInfo = new QueryInfo(BaseTests.queryIdentifier, selectors.size(), BaseTests.hashBitSize, BaseTests.hashKey, BaseTests.dataPartitionBitSize, queryType,
+        false, true, false);
+
+    // Perform the encryption
+    logger.info("Performing encryption of the selectors - forming encrypted query vectors:");
+    EncryptQuery encryptQuery = new EncryptQuery(queryInfo, selectors, paillier);
+    encryptQuery.encrypt(1);
+    logger.info("Completed encryption of the selectors - completed formation of the encrypted query vectors:");
+
+    // Write out files.
+    fileQuerier = File.createTempFile("pir_integrationTest-" + QuerierConst.QUERIER_FILETAG, ".txt");
+    fileQuery = File.createTempFile("pir_integrationTest-" + QuerierConst.QUERY_FILETAG, ".txt");
+
+    localStore.store(fileQuerier.getAbsolutePath(), encryptQuery.getQuerier());
+    localStore.store(fileQuery, encryptQuery.getQuery());
+  }
+
+  private File performDecryption(File responseFile) throws Exception
+  {
+    File finalResults = File.createTempFile("finalFileResults", ".txt");
+    String querierFilePath = fileQuerier.getAbsolutePath();
+    String responseFilePath = responseFile.getAbsolutePath();
+    String outputFile = finalResults.getAbsolutePath();
+    int numThreads = 1;
+
+    Response response = localStore.recall(responseFilePath, Response.class);
+    Querier querier = localStore.recall(querierFilePath, Querier.class);
+
+    // Perform decryption and output the result file
+    DecryptResponse decryptResponse = new DecryptResponse(response, querier);
+    decryptResponse.decrypt(numThreads);
+    decryptResponse.writeResultFile(outputFile);
+    return finalResults;
+  }
+
+}
```

Comment:
This is a cool integration test. As we add more backends, we may need to consider abstract some of the zookeeper and kafka code to a base AbstractPirkIT class and extend that for the different backends.